### ISSUE TITLE
PKG_CONFIG_PATH is incorrectly mangled for pkgconf under Windows

### ISSUE THREAD

**Issue by amyspark** [mesonbuild/meson#10530]
**Describe the bug**

Meson mangles PKG_CONFIG_PATH entries with `:` as the directory separator. However, this is incorrect for the implementation that's shipped by default with MSYS2, which is [pkgconfig](https://github.com/pkgconf/pkgconf). This implementation expects [semicolons](https://github.com/pkgconf/pkgconf/blob/cf48b61a59c1ce6c55a3066a940a6821b9f937bf/libpkgconf/libpkgconf.h#L39-L43) because `:` is the drive letter separator.

According to the Meson log,

```
..\ext_harfbuzz\meson.build:89: WARNING: The "default_options" keyword argument does nothing without a fallback subproject.
Pkg-config binary for MachineChoice.HOST is not cached.
Pkg-config binary missing from cross or native file, or env var undefined.
Trying a default Pkg-config fallback at pkg-config
Found pkg-config: D:\msys64\clang64\bin\pkg-config.EXE (1.8.0)
Determining dependency 'freetype2' with pkg-config executable 'D:\\msys64\\clang64\\bin\\pkg-config.EXE'
env[PKG_CONFIG_PATH]: E:/krita-win/clang64/i_deps/lib/pkgconfig:E:/krita-win/clang64/i_deps/share/pkgconfig
Called `D:\msys64\clang64\bin\pkg-config.EXE --modversion freetype2` -> 1
```

Removing the latter directory, or putting the lib/pkgconfig entry last is sufficient to make it "work" again.

**To Reproduce**

I'm using the [Harfbuzz project](https://github.com/harfbuzz/harfbuzz) with a manual build of Freetype. Then, I am using the following line:

```
meson setup E:\krita-win\clang64\b_deps\ext_freetype\ext_harfbuzz-prefix\src\ext_harfbuzz-build E:\krita-win\clang64\b_deps\ext_freetype\ext_harfbuzz-prefix\src\ext_harfbuzz
            --prefix=E:\krita-win\clang64\i_deps
            --libdir=lib
            -Dbuildtype=optimized
            -Dpkg_config_path=E:/krita-win/clang64/lib/pkgconfig,E:/krita-win/clang64/share/pkgconfig
```

**Expected behavior**

Freetype2 is found.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?: No
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.): Windows 10 21H2
* what Python version are you using e.g. 3.8.0: 3.8.10
* what `meson --version`: 0.62.2
* what `ninja --version` if it's a Ninja build: 1.10.2

**Comment by eli-schwartz** [comment]
I think this was probably fixed via 18bec0d3e31884dc493a03741b2ec9fef17047ee.

Can you try Meson 0.63.0rc1 (you can use `pip install --pre` to install it) and see if the problem is fixed?

**Comment by amyspark** [comment]
Yes, I can confirm it's fixed. Thanks!
